piksi_multi_cpp/include/piksi_multi_cpp/callback.h

```diff
@@ -0,0 +1,50 @@
+#ifndef PIKSI_MULTI_CPP_CALLBACK_H_
+#define PIKSI_MULTI_CPP_CALLBACK_H_
+
+#include <libsbp/sbp.h>
+#include <ros/ros.h>
+#include <memory>
+
+namespace piksi_multi_cpp {
+
+const uint32_t kQueueSize = 100;
+const bool kLatchTopic = true;
+
+// This class handles the callback creation for the different piksi message
+// types.
+class Callback {
```

Comment:
Hm there's a few problems with this class. Callback is quite a non-ideal name, as it is not immediately clear what it does (breaks principle of least surprise).

Also I think it is a bit overkill to create a inherited Callback type for each different message. I would rather group them into one class, e.g.having a PositionCallbackHandler that contains all callbacks and publishers to output positions (I think there are quite  a few of this). Then maybe a DebugCallbackHandler that contains all callbacks  with system states, satelite count etc. and so on. 
And then depending on the receiver type, it will contain more or less of these grouped handlers. I think that would make it a bit more readable

Reply:
I will address this issue in a separate PR as discussed. We will keep the sbp callback handler base class but
a) Try to allow multiple callbacks per message such that individual callback handlers can be defined for the same sbp callback
b) Create inherited superclasses that summarize behavior of similar callbacks, e.g., position callbacks.
c) Template the callback handlers such that they can automatically process